src/lib/logs/log_self_monitoring.py

```diff
@@ -0,0 +1,277 @@
+#     you may not use this file except in compliance with the License.
+#     You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#     Unless required by applicable law or agreed to in writing, software
+#     distributed under the License is distributed on an "AS IS" BASIS,
+#     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#     See the License for the specific language governing permissions and
+#     limitations under the License.
+import asyncio
+import os
+import queue
+import time
+import traceback
+from collections import Counter
+from queue import Queue
+from typing import Dict, List
+
+import aiohttp
+
+from lib.context import LoggingContext, LogsSfmContext, DynatraceConnectivity
+from lib.credentials import create_token, get_dynatrace_log_ingest_url_from_env
+from lib.logs.log_forwarder_variables import SENDING_WORKER_EXECUTION_PERIOD_SECONDS, MAX_MESSAGES_PROCESSED
+from lib.logs.log_sfm_metric_descriptor import LOG_SELF_MONITORING_CONNECTIVITY_METRIC_TYPE, \
+    LOG_SELF_MONITORING_ALL_REQUESTS_METRIC_TYPE, \
+    LOG_SELF_MONITORING_TOO_OLD_RECORDS_METRIC_TYPE, LOG_SELF_MONITORING_PARSING_ERRORS_METRIC_TYPE, \
+    LOG_SELF_MONITORING_PROCESSING_TIME_METRIC_TYPE, LOG_SELF_MONITORING_SENDING_TIME_SIZE_METRIC_TYPE, \
+    LOG_SELF_MONITORING_TOO_LONG_CONTENT_METRIC_TYPE
+from lib.self_monitoring import push_self_monitoring_time_series
+
+
+class LogSelfMonitoring:
+    def __init__(self):
+        self.too_old_records: int = 0
+        self.parsing_errors: int = 0
+        self.records_with_too_long_content: int = 0
+        self.all_requests: int = 0
+        self.dynatrace_connectivity = []
+        self.processing_time_start: float = 0
+        self.processing_time: float = 0
+        self.sending_time_start: float = 0
+        self.sending_time: float = 0
+
+    def calculate_processing_time(self):
+        self.processing_time = (time.perf_counter() - self.processing_time_start)
+
+    def calculate_sending_time(self):
+        self.sending_time = (time.perf_counter() - self.sending_time_start)
+
+
+def aggregate_self_monitoring_metrics(aggregated_sfm: LogSelfMonitoring, sfm_list: List[LogSelfMonitoring]):
+    for sfm in sfm_list:
+        aggregated_sfm.all_requests += sfm.all_requests
+        aggregated_sfm.too_old_records += sfm.too_old_records
+        aggregated_sfm.parsing_errors += sfm.parsing_errors
+        aggregated_sfm.records_with_too_long_content += sfm.records_with_too_long_content
+        aggregated_sfm.dynatrace_connectivity.extend(sfm.dynatrace_connectivity)
+        aggregated_sfm.processing_time += sfm.processing_time
+        aggregated_sfm.sending_time += sfm.sending_time
+    return aggregated_sfm
+
+
+def put_sfm_into_queue(sfm_queue: Queue, sfm: LogSelfMonitoring, logging_context: LoggingContext):
+    try:
+        sfm_queue.put(sfm, True, SENDING_WORKER_EXECUTION_PERIOD_SECONDS + 1)
```

Comment:
I don't think we should block processing/sending workers on `sfm_queue`, we can grind processing to the halt for several seconds this way. Logging exception should be enough